0002-data-collection-extension/README.md

```diff
@@ -1,96 +1,96 @@
+      - [Request](#request)
+      - [Response](#response)
 
-The Data Collection Extension protocol is an endpoint-aware [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
+## Introduction
+Data Collection protocol is an endpoint-aware extension of [Kaa protocol](/0001-kaa-protocol/README.md).
 
-It is intended to solve the problem of transfering collected data to a server for further storing or processing.
+It is designed to collect data from endpoints and transfer it to other services/extensions for storage and/or processing.
 
 ## Language
-
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 
-## Requirements and constraints
-### Problems and possible solutions
-
-1. _Record processing is asynchronous._ Processing might require a significant period of time and may be performed simultaneously or in different order.
-
-2. _The client should know if a record has been delivered and/or processed._ This is needed for client to guarantee data has been processed and free allocated resources.
-
-   Solutions:
-   - Use QoS levels for message delivery confirmation.
-   - Use QoS levels for message processing confirmation.
-
-     MQTT requires all PUBACK messages appear in the same order as received PUBLISH messages. This creates a synchronization point, which conflicts with asynchronous record processing.
-   - Use separate response messages for processing confirmation.
-
-     For CoAP, this is supported by [Separate Response (RFC 7252, Section 5.2.2)](https://tools.ietf.org/html/rfc7252#section-5.2.2).
-
-     For MQTT, this is achieved by publishing a response into the status topic. (See [Kaa Protocol](/0001-kaa-protocol/README.md).)
+The following terms and definitions are used in this RFC.
 
-3. _A server should handle different types of data._ Different endpoints may send data in a variety of formats; the server should know the format to parse the payload.
+- **Record**: a single data point user is interested in.
+- **Batch**: a collection of records uploaded within a single transaction.
+- **Processing confirmation**: an acknowledgment designating that the server has finished processing a request.
 
-   Solutions:
-   - Use distinct resource paths for different formats. Embed `<format_specifier>` into the resource path.
-
-4. _The server should know the endpoint that generates the data._
-
-   Solutions:
-   - Embed `<endpoint_token>` into the resource path.
-
-5. _Little network usage._ Internet connection may be slow, unstable, or costly, so the extension should send as little data as possible.
-
-   Solutions:
-   - Upload data in _batches_. A batch is a number of records, which are uploaded in one network packet.
+## Requirements and constraints
 
-6. _A device may have no ability to generate timestamps._
+Data Collection protocol requirements:
```

Comment:
This section has completely different meaning now and part of the knowledge is lost.